godot_project/autoloads/ui_blocker/ui_blocker.gd

```diff
@@ -0,0 +1,24 @@
+extends CanvasLayer
+
+var _blocker_rect: ColorRect = null
+
+
+func _ready() -> void:
+	_blocker_rect = %BlockerRect as ColorRect
+	_blocker_rect.visible = false
+
+
+func _process(in_delta: float) -> void:
+	_unblock_input()
+
+
+func is_blocking() -> bool:
+	return _blocker_rect.visible
+
+
+func block_input() -> void:
```

Comment:
nit: when calling this function from outside I would expect it's permanent operation until I will manually unblock, maybe we could rename it to `block_current_frame_inputs` ?

Reply:
Done!